frontend/src/modules/notification/actions.tsx

```diff
@@ -0,0 +1,60 @@
+import React from "react"
+import { AppDispatch, RootState } from "../../store"
+import { NOTIFICATION_REMOVE, NotificationAction as NotificationAction, NotificationID, NotificationProps, NotificationRemoveAction as NotificationRemoveAction, NOTIFICATION, NotificationType } from "./models"
+import { notification } from "antd"
+
+type AntdNotificationAPI = keyof Pick<typeof notification, 'success' | 'info' | 'warning' | 'error'>
+
+// converts our NotificationType to the function names in the Antd Notification API
+const convertToAntdNotificationAPI: { [key in NotificationType]: AntdNotificationAPI } = {
+    [NotificationType.SUCCESS]: 'success',
+    [NotificationType.INFO]: 'info',
+    [NotificationType.WARNING]: 'warning',
+    [NotificationType.ERROR]: 'error'
+}
+
+const hasNotification = (state: RootState, id: NotificationID) => {
+    return state.notifications.some((notification) => notification.id === id)
+}
+
+export const notify = (id: NotificationID, props: NotificationProps) => (dispatch: AppDispatch, getState: () => RootState) => {
+    if (hasNotification(getState(), id)) {
+        dispatch(closeNotification(id))
+    }
+
+    notification[convertToAntdNotificationAPI[props.type]]({
+        message: props.title,
+        description: <pre style={{ fontSize: '0.8em', whiteSpace: 'pre-wrap' }}>{props.description}</pre>,
+        duration: props.duration ?? 4.5, // 4.5 is the default duration in antd
```

Comment:
Maybe you should not shadow antd default duration by setting 4.5 but instead leave the duration undefined so the default takes over ?

Reply:
If antd changes the default it might lead to baffled devs trying to figure why the duration is 4.5 and not the default ;). I typically try not to override a default if i can. It is not a fault though.